activerecord/test/cases/adapters/postgresql/schema_test.rb

```diff
@@ -780,3 +780,46 @@ def test_schema_dumps_index_included_columns
     end
   end
 end
+
+class SchemaIndexNullsNotDistinctTest < ActiveRecord::PostgreSQLTestCase
+  include SchemaDumpingHelper
+
+  setup do
+    @connection = ActiveRecord::Base.connection
+    @connection.create_table "trains" do |t|
+      t.string :name
+    end
+  end
+
+  teardown do
+    @connection.drop_table "trains", if_exists: true
+  end
+
+  def test_nulls_not_distinct_is_dumped
+    if supports_nulls_not_distinct?
```

Comment:
A nitpick but generally it's preferable not to define test at all if we know we aren't going to run anything and make any assertions. An empty test unnecessarily gets executed and contributes to run statistics while not testing anything. 

Often we may need to be inside the test execution in order to decide whether we can assert anything. In this case I'd prefer the test being explicitly reported as skipped rather than as "successfully ran" 

```suggestion
    skip("current adapter doesn't support nulls not distrinct") unless supports_nulls_not_distinct?
```

Reply:
Love it. Most of this PR I tried to use established patterns (postgres adapter is rife with `if supports_*`), but this is clearly better. 😄